commit 55e357bc316bc3f46e96bc55415cb7d512c8db6f
Author: Thomas Backlund <tmb@mageia.org>
Date:   Tue Nov 14 12:37:51 2017 +0200

    iwlwifi: fix firmware names for 9000 and A000 series hw
    
    commit c2c48ddfc8b03b9ecb51d2832b586497b37531bc upstream.
    
    iwlwifi 9000 and a0000 series hw contains an extra dash in firmware
    file name as seeen in modinfo output for kernel 4.14:
    
    firmware:       iwlwifi-9260-th-b0-jf-b0--34.ucode
    firmware:       iwlwifi-9260-th-a0-jf-a0--34.ucode
    firmware:       iwlwifi-9000-pu-a0-jf-b0--34.ucode
    firmware:       iwlwifi-9000-pu-a0-jf-a0--34.ucode
    firmware:       iwlwifi-QuQnj-a0-hr-a0--34.ucode
    firmware:       iwlwifi-QuQnj-a0-jf-b0--34.ucode
    firmware:       iwlwifi-QuQnj-f0-hr-a0--34.ucode
    firmware:       iwlwifi-Qu-a0-jf-b0--34.ucode
    firmware:       iwlwifi-Qu-a0-hr-a0--34.ucode
    
    Fix that by dropping the extra adding of '"-"'.
    
    Signed-off-by: Thomas Backlund <tmb@mageia.org>
    Signed-off-by: Luca Coelho <luciano.coelho@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/wireless/intel/iwlwifi/cfg/9000.c b/drivers/net/wireless/intel/iwlwifi/cfg/9000.c
index e8b5ff42f5a8..6f43faab4910 100644
--- a/drivers/net/wireless/intel/iwlwifi/cfg/9000.c
+++ b/drivers/net/wireless/intel/iwlwifi/cfg/9000.c
@@ -79,11 +79,11 @@
 #define IWL9000_MODULE_FIRMWARE(api) \
 	IWL9000_FW_PRE "-" __stringify(api) ".ucode"
 #define IWL9000RFB_MODULE_FIRMWARE(api) \
-	IWL9000RFB_FW_PRE "-" __stringify(api) ".ucode"
+	IWL9000RFB_FW_PRE __stringify(api) ".ucode"
 #define IWL9260A_MODULE_FIRMWARE(api) \
-	IWL9260A_FW_PRE "-" __stringify(api) ".ucode"
+	IWL9260A_FW_PRE __stringify(api) ".ucode"
 #define IWL9260B_MODULE_FIRMWARE(api) \
-	IWL9260B_FW_PRE "-" __stringify(api) ".ucode"
+	IWL9260B_FW_PRE __stringify(api) ".ucode"
 
 #define NVM_HW_SECTION_NUM_FAMILY_9000		10
 
diff --git a/drivers/net/wireless/intel/iwlwifi/cfg/a000.c b/drivers/net/wireless/intel/iwlwifi/cfg/a000.c
index a440140ed8dd..7eade165b747 100644
--- a/drivers/net/wireless/intel/iwlwifi/cfg/a000.c
+++ b/drivers/net/wireless/intel/iwlwifi/cfg/a000.c
@@ -80,15 +80,15 @@
 #define IWL_A000_HR_A0_FW_PRE	"iwlwifi-QuQnj-a0-hr-a0-"
 
 #define IWL_A000_HR_MODULE_FIRMWARE(api) \
-	IWL_A000_HR_FW_PRE "-" __stringify(api) ".ucode"
+	IWL_A000_HR_FW_PRE __stringify(api) ".ucode"
 #define IWL_A000_JF_MODULE_FIRMWARE(api) \
-	IWL_A000_JF_FW_PRE "-" __stringify(api) ".ucode"
+	IWL_A000_JF_FW_PRE __stringify(api) ".ucode"
 #define IWL_A000_HR_F0_QNJ_MODULE_FIRMWARE(api) \
-	IWL_A000_HR_F0_FW_PRE "-" __stringify(api) ".ucode"
+	IWL_A000_HR_F0_FW_PRE __stringify(api) ".ucode"
 #define IWL_A000_JF_B0_QNJ_MODULE_FIRMWARE(api) \
-	IWL_A000_JF_B0_FW_PRE "-" __stringify(api) ".ucode"
+	IWL_A000_JF_B0_FW_PRE __stringify(api) ".ucode"
 #define IWL_A000_HR_A0_QNJ_MODULE_FIRMWARE(api) \
-	IWL_A000_HR_A0_FW_PRE "-" __stringify(api) ".ucode"
+	IWL_A000_HR_A0_FW_PRE __stringify(api) ".ucode"
 
 #define NVM_HW_SECTION_NUM_FAMILY_A000		10
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1074_55e357bc316bc3f46e96bc55415cb7d512c8db6f.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1074_55e357bc316bc3f46e96bc55415cb7d512c8db6f.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
